impl/client/src/main/javascript/web/pentaho/visual/color/paletteProperty.js

```diff
@@ -0,0 +1,218 @@
+/*!
+ * Copyright 2010 - 2017 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "module",
+  "pentaho/i18n!messages",
+  "pentaho/type/ValidationError",
+  "pentaho/type/util",
+  "pentaho/util/object",
+  "pentaho/util/error"
+], function(module, bundle, ValidationError, typeUtil, O, error) {
+
+  "use strict";
+
+  return [
+    "property",
+    "pentaho/visual/color/level",
+    "pentaho/visual/color/palette",
+    {$instance: {type: ["pentaho/visual/color/palette"]}},
```

Comment:
Why is `allPalettesList` needed? Is it to ensure all palettes are loaded?

Reply:
Yep, this is a new module dependency syntax for loading instances. Because it is a list type that is being requested, all registered instance palettes are pre-loaded. The class code below assumes that every registered palette can be found synchronously.